### ISSUE TITLE
Cronjob v2 controller bugs：jobs will be pending after when update cronjob 

### ISSUE THREAD

**Issue by yanhongchang** [kubernetes/kubernetes#112092]
### What happened?

sometimes， the jobs will stop create and Perform after when I update the cronjob,  such as update the image.   

### What did you expect to happen?

I think the update about cronjob resource will not affect the job.

### How can we reproduce it (as minimally and precisely as possible)?

This is an occasional problem. It doesn't happen every time.

### Anything else we need to know?

I noticed the problem cronjob status is different about other normals

** the problem cronjob：**

``
apiVersion: batch/v1beta1
kind: CronJob
metadata:
  creationTimestamp: "2022-08-16T07:37:15Z"
 ......
status:
  active:
  - apiVersion: batch/v1
    kind: Job
  lastScheduleTime: "2022-08-17T02:05:00Z"
``

**the normal cronjob:**

`
apiVersion: batch/v1beta1
kind: CronJob
metadata:
  creationTimestamp: "2022-08-17T02:36:22Z"
  ......
status:
  active:
  - apiVersion: batch/v1
    kind: Job
    name: 5minute-27695895
    namespace: test
    resourceVersion: "172518042"
    uid: 1484a052-fc8c-42cd-96db-7193c7952566
  lastScheduleTime: "2022-08-29T06:15:00Z"
`

**OR**

apiVersion: batch/v1beta1
kind: CronJob
metadata:
  creationTimestamp: "2022-08-17T02:36:22Z"
  ......
status:
  lastScheduleTime: "2022-08-29T06:15:00Z"

### Kubernetes version

<details>

```console
$ kubectl version

Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.8", GitCommit:"5575935422cc1cf5169dfc8847cb587aa47bac5a", GitTreeState:"clean", BuildDate:"2021-06-16T13:00:45Z", GoVersion:"go1.15.13", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.8", GitCommit:"5575935422cc1cf5169dfc8847cb587aa47bac5a", GitTreeState:"clean", BuildDate:"2021-06-16T12:53:07Z", GoVersion:"go1.15.13", Compiler:"gc", Platform:"linux/amd64"}

```

</details>

### Cloud provider

<details>
none 
</details>

### OS version

<details>

```console

$ cat /etc/os-release
# Alibaba Cloud Linux 3 (Soaring Falcon)

$ uname -a
# 5.10.84-10.4.al8.x86_64 #1 SMP Tue Apr 12 12:31:07 CST 2022 x86_64 x86_64 x86_64 GNU/Linux



</details>


### Install tools

<details>
kubeadm
</details>


### Container runtime (CRI) and version (if applicable)

<details>
docker

Client: Docker Engine - Community
 Version:           20.10.8
 API version:       1.41
 Go version:        go1.16.6
 Git commit:        3967b7d
 Built:             Fri Jul 30 19:53:39 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.8
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.6
  Git commit:       75249d8
  Built:            Fri Jul 30 19:52:00 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.5.11
  GitCommit:        3df54a852345ae127d1fa3092b95168e4a88e2f8
 runc:
  Version:          1.0.3
  GitCommit:        v1.0.3-0-gf46b6ba
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
</details>


### Related plugins (CNI, CSI, ...) and versions (if applicable)

_No response_

**Comment by yanhongchang** [comment]
> 

I have read the part of cronjob controllers v2 code both v1.20 and v.122 even master branch, there is no much different between those branch, I think the bugs still exists in latest version. I will try to fixed the problem today. give me some times.

**Comment by yanhongchang** [comment]
> > $ kubectl version
> 
> Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.8", GitCommit:"5575935422cc1cf5169dfc8847cb587aa47bac5a", GitTreeState:"clean", BuildDate:"2021-06-16T13:00:45Z", GoVersion:"go1.15.13", Compiler:"gc", Platform:"linux/amd64"} Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.8", GitCommit:"5575935422cc1cf5169dfc8847cb587aa47bac5a", GitTreeState:"clean", BuildDate:"2021-06-16T12:53:07Z", GoVersion:"go1.15.13", Compiler:"gc", Platform:"linux/amd64"}
> 
> < 1.22 is out of support. does it work with newer k8s?


I have read the part of cronjob controllers v2 code both v1.20 and v.122 even master branch, there is no much different between those branch, I think the bugs still exists in latest version. I will try to fixed the problem today. give me some times.

**Comment by pacoxu** [comment]
Link the pr https://github.com/kubernetes/kubernetes/pull/112116

**Comment by yanhongchang** [comment]
> Link the pr #112116

this pr is closed， the new one pr link #112275

**Comment by yanhongchang** [comment]
@pacoxu Hi Pacoxu，I need you help, Is there any problem with my pr #112275, why Why did the reviewers do not reply? or am I missing any step for my pr？ thanks！
